force-app/test/ContactInsertTests_TEST.cls

```diff
@@ -0,0 +1,41 @@
+@IsTest
+private class ContactInsertTests_TEST {
+
+    @IsTest
+    static void insert_duplicate_contact_should_not_throw_required_field_missing_error() {
+        // Arrange
+        String exceptionMessage;
+        Contact existingContact = TEST_ContactBuilder.dummyContact();
+        existingContact.MailingStreet = '123 Main';
+        existingContact.MailingCity = 'ABCity';
+        insert existingContact;
+        Contact dupeContact = TEST_ContactBuilder.dummyContact();
+        dupeContact.MailingStreet = '123 Main';
+        dupeContact.MailingCity = 'ABCity';
+
+        // Act
+        try {
+            // This will succeed or fail depending on the org's duplicate rule configuration.
+            // If it succeeds the test does and should pass.
+            insert dupeContact;
+        } catch (Exception e) {
+            exceptionMessage = e.getMessage();
+        }
+
+        // Assert
+        if (exceptionMessage != null) {
+            // If it fails, assert that it is not due to the required fields missing error
+            System.assertEquals(
+                    false,
```

Comment:
Not sure if we can prevent this, but could there be a chance that the error message for missing required fields might change in the future? If it does, this `assert` will fail

Reply:
In this case I don't think the assert would fail, it just wouldn't flag for us a regression. If the assert message is changed by the platform, then this test wouldn't fail even if somebody change the code back to reintroduce the regression. However I think that is a risk that is acceptable in order to give us a test that will flag a regression if a change is made to go back and attempt to insert addresses without account ids here.  See this section in the design doc: https://salesforce.quip.com/HO0zAflm4oyr#temp:C:PGB20c4f7a7cbcf9468e5a5a3967